docs/dynamic-bundling/index.md

```diff
@@ -0,0 +1,41 @@
+# Dynamic Bundling
+
+One way to address the gap in serving primitives is a technique we’ll call “dynamic bundling” here.
+Traditional asset deployments require that there’s a strict 1-to-1 mapping between output files generated at build time and network requests at runtime.
+The output files of the build process are served by a simple file server without any additional transformations beyond compression.
+
+Dynamic bundling adds a runtime component that allows for assembly of network responses from file fragments (“chunks”) generated at build time.
+This makes using a simple file server impossible but offers the following advantages:
+
+* Reduced network requests caused by code splitting.
+  Very fine-grained splitting strategies can be used without the overhead and latency anomalies that come with tens or hundreds of individual requests.
+* Reduced number of output files (and build runtime) when generating many variants.
+  E.g. each use of a feature flag (or experiment) can be isolated into one small chunk.
+  The correct code fragment can then be dynamically selected while assembling the bundle.
+  For `n` feature flags, we only need `O(n)` files.
+    - Build-time assembly, on the other hand, is forced to combine files that use many different feature flags,
+      which means generating `O(2n)` output files for `n` feature flags.
```

Comment:
How did you come up with `O(2n)` here?

Reply:
I believe it should be 2^n :)

Reply:
Heh, good catch! Copy&pasted from a google doc that used 2<sup>n</sup> which got lost in translation.